custom/ucr_ext/tests.py

```diff
@@ -0,0 +1,382 @@
+import uuid
+from xml.etree import ElementTree
+from datetime import date, datetime
+from django.test import SimpleTestCase, TestCase
+from corehq.apps.userreports.expressions.factory import ExpressionFactory
+from corehq.apps.userreports.specs import EvaluationContext
+from corehq.apps.receiverwrapper.util import submit_form_locally
+from casexml.apps.case.mock import CaseStructure, CaseFactory
+from casexml.apps.case.tests.util import delete_all_cases, delete_all_xforms
+
+
+def _safe_text(input_value):
+    if input_value is None:
+        return ''
+    try:
+        return str(input_value)
+    except:
+        return ''
+
+
+def _create_element_with_value(element_name, value):
+    elem = ElementTree.Element(element_name)
+    elem.text = _safe_text(value)
+    return elem
+
+
+class DiffCalendarMonthsExpressionTest(SimpleTestCase):
+    def test_cases(self):
+        expression = ExpressionFactory.from_spec({
+            "type": "ext_diff_calendar_months",
+            "from_date_expression": {
+                "type": "property_name",
+                "property_name": "from",
+            },
+            "to_date_expression": {
+                "type": "property_name",
+                "property_name": "to",
+            },
+        })
+        self.assertEqual(0, expression({"from": "2015-06-03", "to": "2015-06-06"}))
+        self.assertEqual(1, expression({"from": "2015-06-29", "to": "2015-07-01"}))
+        self.assertEqual(12, expression({"from": "2015-06-29", "to": "2016-06-06"}))
+        self.assertEqual(11, expression({"from": "2015-06-29", "to": "2016-05-01"}))
+        self.assertEqual(15, expression({"from": "2015-06-29", "to": "2016-09-06"}))

```

Comment:
might be good to add a test for the case where "from" comes after "to" which would also serve to document the expected behavior